### PR TITLE
some suggestions

### PR DESCRIPTION
**Pull request by moinmoin-sh**
- remove static vars  : getInputType(), getEventType()
- replace multiple defaults with preset :  getEventType()
- discovered possible buffer overflow  : getConfigTopic(), getEventTopic() 

this PR has 2 suggestions (please see the change log)
and a hint to a possible buffer overflow condition (please see the large comment block in getConfigTopic() ). No solution, comment only. Many solutions possible - your choice ...

### CODE REVIEW

File: Firmware/USMArduinoFirmware/USMArduinoFirmware.ino [hunk 1]
```diff
@@ -482,13 +481,12 @@ char * getInputType(uint8_t type)
...
   }
-  return inputType;
 }
 
-char * getEventType(uint8_t type, uint8_t state)
+void getEventType(char eventType[], uint8_t type, uint8_t state)
 {
   // Determine what event we need to publish
-  static char eventType[7];
+  sprintf_P(eventType, PSTR("ERROR"));  // preset if no matching condition found
```
Comment (sumnerboy12) [inline Firmware/USMArduinoFirmware/USMArduinoFirmware.ino:489]: I did consider this, but what happens if the type is `UP` - will the `eventType` array be returned with `UPROR`? Or will the null termination ensure that we only see `UP`?

File: Firmware/USMArduinoFirmware/USMArduinoFirmware.ino [hunk 2]
```diff
@@ -577,6 +559,24 @@ char * getConfigTopic(char topic[])
   }
   else
   {
+    // Attention :
+    // if
+    //  const char *  mqtt_client_id        = NULL;       // Max 15 chars, optional
```
Comment (sumnerboy12) [inline Firmware/USMArduinoFirmware/USMArduinoFirmware.ino:564]: Good catch - this array was sized when we were using a fixed 6 chars for the device id - I think we need to support a reasonably large base topic and client id - for those that have nested topic structures etc.

So I think increase the topic array size. I think the max size will be 40 chars - so lets make it 42?

File: Firmware/USMArduinoFirmware/USMArduinoFirmware.ino (hunk 1 above)
Comment (sumnerboy12) [inline Firmware/USMArduinoFirmware/USMArduinoFirmware.ino:489]: Sorry to be pedantic but I am not a fan of inline comments :) Could you remove please?

**Review by sumnerboy12 — COMMENTED** [review]
Thanks Moin - these look good and that is a great catch on the topic array. Really appreciate you reviewing my code!!

File: Firmware/USMArduinoFirmware/USMArduinoFirmware.ino (hunk 1 above)
Comment (moinmoin-sh) [reply Firmware/USMArduinoFirmware/USMArduinoFirmware.ino]: sprintf works on uninitialized buffers, so this is covered

File: Firmware/USMArduinoFirmware/USMArduinoFirmware.ino (hunk 2 above)
Comment (moinmoin-sh) [reply Firmware/USMArduinoFirmware/USMArduinoFirmware.ino]: set topic[42]

File: Firmware/USMArduinoFirmware/USMArduinoFirmware.ino (hunk 1 above)
Comment (moinmoin-sh) [reply Firmware/USMArduinoFirmware/USMArduinoFirmware.ino]: removed, was meant to be an explanation why I did this

**Review by sumnerboy12 — APPROVED** [review]
Looks good, thanks again!

**Comment by moinmoin-sh** [comment]
you are faster merging than I can write 👍

**Comment by sumnerboy12** [comment]
Ha, sitting in the kids bedroom waiting for them to go to sleep so did it
on my phone!

**Pull request merged**